Blazor.ExtraDry/Blazor.ExtraDry.Core.Tests/Rules/RuleEngineDeleteTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Blazor.ExtraDry.Core.Tests.Rules {
+    public class RuleEngineDeleteTests {
+
+        [Fact]
+        public void DeleteRequiresItem()
+        {
+            var rules = new RuleEngine(new ServiceProviderStub());
+
+            Assert.Throws<ArgumentNullException>(() => rules.Delete((object)null, NoOp));
+        }
+
+        [Fact]
+        public void DeleteSoftDeletesByDefault()
+        {
+            var rules = new RuleEngine(new ServiceProviderStub());
```

Comment:
I love the explicit Arrange, Act Assert pattern here, makes consumption easier for other devs.